### DISCUSSION TITLE
Selective sync; adding directory from ODFB to OD sync

### DISCUSSION THREAD

**Discussion by JosN1** [abraunegg/onedrive#1638] [Q&A] [answered]
Currently, my installation is working well at synchronizing one specific folder from my consumer OneDrive account. I would like to add two specific folders from my OneDrive for Business account (in M365). I cannot figure out how to do that. The syncfile has only the name of the currently synchronized folder, not the entire URI or name of the OD service, so it would do no good to simply add the ODFB folder names. How can I add a new URI to my synclist? 

Also, big thanks to those who are maintaining this app and these files. It's all extremely useful.

**Comment by abraunegg** [comment, marked as answer]
@JosN1 
Unfortunately you cannot mix OneDrive Personal & OneDrive Business account types for syncing. The authentication methods are different, the authentication tokens also cannot be used with the other service. The way which the Windows client does this, is that it runs 2 separate clients, one for each account type. The same is true for this Linux client.

Please follow this document to configure separate account configurations for OneDrive Personal & OneDrive Business: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-mulitple-onedrive-accounts--configurations

Now - most likely you want to then store all your data under a single directory. Depending on your configuration, this is what I suggest:
* Configure your OneDrive Personal 'sync_dir' to: `/somewhere_other_than_your_user_home_directory/onedrive-personal/`
* Configure your OneDrive Business 'sync_dir' to: `/somewhere_other_than_your_user_home_directory/onedrive-business/`
* Symbolic link then `/somewhere_other_than_your_user_home_directory/onedrive-personal/` -> `~/OneDrive/Personal/`
* Symbolic link then `/somewhere_other_than_your_user_home_directory/onedrive-business/` -> `~/OneDrive/Business/`

This has the effect then of when your look in your home directory under OneDrive, all your data is there and usable.

**Reply by JosN1** [reply]
@abraunegg Thanks for the prompt response. I'll take a look at that resource. The configuring might be pushing my embryonic Linux skills, but I'm looking forward to the effort.
